src/solver/constraints-builder/cbuilder.cpp

```diff
@@ -91,9 +91,28 @@ bool CBuilder::update(bool applyCheckBox)
 
     for (auto linkInfoIt = pLink.begin(); linkInfoIt != pLink.end(); linkInfoIt++)
     {
-        // Try to open the file
-        if (not (*linkInfoIt)->ptr->loadDataFromCSVfile(Matrix<>::optImmediate))
+        auto linkInfo = *linkInfoIt;
+        Data::AreaLink* link = linkInfo->ptr;
+
+        // Try to open link data files
+        // ... Load parameters from a data file 
+        YString dataFilename = link->parameters.jit->sourceFilename;
+        if (not link->parameters.loadFromCSVFile(dataFilename, Data::fhlMax, HOURS_PER_YEAR, Matrix<>::optImmediate))
+        {
+            return false;
+        }
+        // ... Load direct capacities from a data file 
+        dataFilename = link->directCapacities.jit->sourceFilename;
+        if (not link->directCapacities.loadFromCSVFile(dataFilename, Data::fhlMax, HOURS_PER_YEAR, Matrix<>::optImmediate))
+        {
+            return false;
+        }
+        // ... Load indirect capacities from a data file 
+        dataFilename = link->indirectCapacities.jit->sourceFilename;
+        if (not link->indirectCapacities.loadFromCSVFile(dataFilename, Data::fhlMax, HOURS_PER_YEAR, Matrix<>::optImmediate))
+        {
```

Comment:
This will fail on older studies. Require study upgrade before running the Kirchoff generator ?

Reply:
Or maybe check the study version number to use the legacy format when necessary ?

Reply:
Create common loading function used also in `CBuilder::deletePreviousConstraints`

Reply:
ok, done